central/complianceoperator/v2/report/manager/watcher/scanwatcher.go

```diff
@@ -60,6 +61,17 @@ type ScanWatcherResults struct {
 	Error        error
 }
 
+// DeleteOldResults associated with the Scan of the given ScanWatcherResults
+func DeleteOldResults(ctx context.Context, results *ScanWatcherResults, store resultsDS.DataStore) error {
```

Comment:
Where is this used? I see only usages of the `ds.DeleteOldResults`.
This also seems to be untested - at least I cannot find the test code in this PR-diff.

Are we protected against providing `results` as nil?
Could this func maybe be a member of type `ScanWatcherResults`?